### PR TITLE
Added Role UI BDD feature file

### PR DESCRIPTION
**Pull request by pcardiga**
BDD scenarios for Role UI (list, view, CRUD) - https://jira.ez.no/browse/EZP-24685

Todo:
- [x] Implementation

Later follow up:
- Validation of Role limitations view with links and human readable names
- Role assignment
- Role versioning
- Role list pagination

### CODE REVIEW

**Comment by pcardiga** [comment]
ping @andrerom , @miguelcleverti , @micszo 
Please review

**Comment by micszo** [comment]
Shouldn't the last scenario before "Deleting Roles" be called "cancel the update of a role"? :)

**Comment by pcardiga** [comment]
you are right @micszo , there was a typo there.

File: Features/RoleUI/RoleUI.feature [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    And I should see a "Save" button
+    And I should see a "Cancel" button
+
+  @javascript @common
+  Scenario: Update a role
+    Given I have an "Organizers" role
+    And I edit the "Organizers" role
+    And I set "old_Organizers" as the "Name"
+    When I click on "Save" button
+    Then the Role is successfully published
```
Comment (andrerom) [inline Features/RoleUI/RoleUI.feature]: these published / not published could also instead verify presence / non presence of the Role list afterwards right? (depends a bit on how @glye  setup redirects)

File: Features/RoleUI/RoleUI.feature [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    And I create a new role
+    And I set "Organizers" as the "Name"
+    When I click on "Save" button
+    Then I see a message saying that the name already exists
+    And the Role is not published
+
+  @javascript @common
+  Scenario: cancel the creation of a role
+    Given I am on the Roles page
+    And I create a new role
```
Comment (andrerom) [inline Features/RoleUI/RoleUI.feature]: Is it intentional that role is sometimes not written with capital R?

File: Features/RoleUI/RoleUI.feature [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      | Module | Function | Limitation |
+    And I should see an "Edit" button
+    And I should see a "Delete" button
+    And I should see a label for the Role Assigments
+    And I should see a group with the Role Assigments
+      | User/group | Limitation |
+
+  @javascript @edge
+  Scenario: Display the information of a role with empty policies
+    Given I have an "Organizers" role
```
Comment (andrerom) [inline Features/RoleUI/RoleUI.feature]: This sentence is not clear, when I first read it I assumed it meant Administrator had this role (assigned). Because `I am an Administrator`, so `I have <role>` basically means it is assigned to me :)

But what is being implied here from the use is that it exist, priorly created, so maybe rephrase this sentence to more clearly relfect that.

File: Features/RoleUI/RoleUI.feature [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  ##
+  #
+  # Deleting Roles
+  #
+  ##
+  @javascript @common
+  Scenario: Delete a role
+    Given I have an "Organizers" role
+    When I delete the "Organizers" role
+    Then the Role is successfully deleted
```
Comment (andrerom) [inline Features/RoleUI/RoleUI.feature]: same here as published, could just as well be a sentence verifying it is no longer present in list of roles, would be more clear as to how you expect the change to affect the application.

File: Features/RoleUI/RoleUI.feature (hunk 3 above)
Comment (pcardiga) [inline Features/RoleUI/RoleUI.feature]: The objective is to have an "Organizers" role as pre-condition for the test action.
Maybe ``` Given an "Organizers" role exists ``` fits better the purpose.

File: Features/RoleUI/RoleUI.feature (hunk 2 above)
Comment (pcardiga) [inline Features/RoleUI/RoleUI.feature]: It was a way to differentiate different conceptual things: the name of a place (the "R|r"oles page) from a name of an object (the role itself). But i see this decision is raising doubts, so it's not as clear as i tough when i wrote the sentences. Being so, it might be better to level everything and use smallness in both cases.

File: Features/RoleUI/RoleUI.feature (hunk 1 above)
Comment (pcardiga) [inline Features/RoleUI/RoleUI.feature]: Yes it can. In BDD discussions apart from the current one, it was decided to leave the "publish" action in separate scenarios from the "view" action.

File: Features/RoleUI/RoleUI.feature (hunk 3 above)
Comment (andrerom) [inline Features/RoleUI/RoleUI.feature]: yes, something like that is more clear

File: Features/RoleUI/RoleUI.feature (hunk 2 above)
Comment (andrerom) [inline Features/RoleUI/RoleUI.feature]: Let it be for now, I need to think about it as well :)

File: Features/RoleUI/RoleUI.feature (hunk 1 above)
Comment (andrerom) [inline Features/RoleUI/RoleUI.feature]: ok

**Comment by andrerom** [comment]
Other then open questions/discussions: +1

File: Features/RoleUI/RoleUI.feature (hunk 3 above)
Comment (glye) [inline Features/RoleUI/RoleUI.feature]: /me agrees.

File: Features/RoleUI/RoleUI.feature (hunk 1 above)
Comment (glye) [inline Features/RoleUI/RoleUI.feature]: Save/cancel/delete actions redirect to role list, yes.

**Comment by glye** [comment]
+1

**Comment by pcardiga** [comment]
Discussed questions added to scenarios:
- Updated sentence ``` Given I have an "Organizers" role ```to ``` Given an "Organizers" role exists ```
- Added a verification sentence in scenarios after the updating/deleting/cancelling actions

**Comment by andrerom** [comment]
Ok, so ready for implementation I guess.

In addition to this we need:
- PR on ezplatform adding the new profile / scenarios
- Change travis testing PR to reflect new profile / scenarios to test

And Qa will need to give us heads up when everything is passing locally so we can safly merge all :)

**Comment by glye** [comment]
I just realised there is no pagination in the role list. We need that, and BDD for it.

**Comment by andrerom** [comment]
true, but different story, moved it to backlog and marked that it will be fixed in eZ Platform.

**Comment by miguelcleverti** [comment]
@andrerom implementations added

**Comment by andrerom** [comment]
@miguelcleverti this branch is rebased/merged wrong, seems this PR needs to be reopened against master as the upstream PR is merged by now.

**Comment by miguelcleverti** [comment]
@andrerom maybe it is better to just colsed this PR and I will create a new branch and cherry-pick the commits and open a new PR?

**Comment by andrerom** [comment]
> @andrerom maybe it is better to just colsed this PR and I will create a new branch and cherry-pick the commits and open a new PR?

yes this is the simplest way.

**Comment by miguelcleverti** [comment]
New PR #343

**Comment by pcardiga** [comment]
Closed in favor of PR [#343](https://github.com/ezsystems/PlatformUIBundle/pull/343)
